clicommand/env.go

```diff
@@ -32,20 +35,41 @@ var EnvCommand = cli.Command{
 			Usage:  "Pretty print the JSON output",
 			EnvVar: "BUILDKITE_AGENT_ENV_PRETTY",
 		},
+		cli.BoolFlag{
+			Name:  "from-env-file",
+			Usage: "Source environment from file described by $BUILDKITE_ENV_FILE",
+		},
+		cli.StringFlag{
+			Name:  "print",
+			Usage: "Print a single environment variable by `NAME` as raw text followed by a newline",
+		},
 	},
 	Action: func(c *cli.Context) error {
-		env := os.Environ()
-		envMap := make(map[string]string, len(env))
+		var envMap map[string]string
+		var err error
 
-		for _, e := range env {
-			k, v, _ := strings.Cut(e, "=")
-			envMap[k] = v
+		if c.Bool("from-env-file") {
+			envMap, err = loadEnvFile(os.Getenv("BUILDKITE_ENV_FILE"))
```

Comment:
This feels weird to me. I expect `--from-env-file` to take a value, but default to an environment variable, like our other options. But if we defaulted to the env file then it would break dumping of the current env as json.

Reply:
Yep. I like your idea of moving the current “dump entire process env as JSON” use-cases into a subcommand/flag(s), which then frees up the `buildkite-agent env` namespace to support some other environment-related use-cases.